File: app/src/main/java/com/selflearning/chemistree/chemistry/elements/ElementDao.java
package com.selflearning.chemistree.chemistry.elements;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

import selflearning.chemistree.domain.chemistry.elements.Element;

//@Dao
//public interface ElementDao {
//
//    @Query("SELECT * FROM elements")
//    List<Element> getAllElements();
//
//    @Query("SELECT * FROM elements WHERE atomicNumber = :id")
//    Element getElementById(int id);
//
//    @Insert(onConflict = OnConflictStrategy.REPLACE)
//    void insertAll(List<Element> elements);
//}

File: app/src/main/java/com/selflearning/chemistree/domain/chemistry/elements/ElementDao.kt
package com.selflearning.chemistree.domain.chemistry.elements

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import kotlinx.coroutines.flow.Flow

@Dao
interface ElementDao {

    @Query("SELECT * FROM element")
    fun getAll(): Flow<List<Element>>

    @Query("SELECT * FROM element WHERE atomicNumber = :id")
    fun getElementById(id: Int): Element

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(elements: List<Element>)
}